Fix for https://github.com/Creators-of-Create/Create/issues/7250

dan96kid opened the pull request:
"Fine, I'll do it myself." - Thanos

This is a fix for https://github.com/Creators-of-Create/Create/issues/7250 that takes text on the front and back of signs into account

dan96kid committed:
"Fine, I'll do it myself"

Fix for https://github.com/Creators-of-Create/Create/issues/7250 that should hopefully not break anything

dan96kid committed:
"Fine, I'll do it myself."

Fix for https://github.com/Creators-of-Create/Create/issues/7250 that should hopefully not break anything. Takes the text on the front and back of signs into account.

VoidLeech commented on src/main/java/com/simibubi/create/content/redstone/displayLink/target/SignDisplayTarget.java:
```diff
@@ -20,22 +20,25 @@ public void acceptText(int line, List<MutableComponent> text, DisplayLinkContext
 				break;
 
-			signText = signText.setMessage(i + line, text.get(i));
+			//signText = signText.setMessage(i + line, text.get(i));
```
This can just be removed.

dan96kid replied:
I agree.

VoidLeech commented on src/main/java/com/simibubi/create/content/redstone/displayLink/target/SignDisplayTarget.java:
```diff
@@ -20,22 +20,25 @@ public void acceptText(int line, List<MutableComponent> text, DisplayLinkContext
-				sign.setText(signText, side);
-		context.level()
-			.sendBlockUpdated(context.getTargetPos(), sign.getBlockState(), sign.getBlockState(), 2);
+			context.level().sendBlockUpdated(context.getTargetPos(), sign.getBlockState(), sign.getBlockState(), 2);
```
You should set the text before telling the level the block got updated.

dan96kid replied:
Okay.

I am now wondering... Do both sides of the sign need their own `sendBlockUpdate()`, or can we get away with only one?

VoidLeech replied:
One is enough. The previous code only called it once. And please wrap the part of the changes after if (changed) into {}

dan96kid committed:
Updated SignDisplayTarget.java with VoidLeech's suggestions

dan96kid committed:
Updated SignDisplayTarget.java again

- Removed unused import statement.
- Wrapped all code in the if statements with {}.
- Removed a seemingly unnecessary `sendBlockUpdated`.

Rafacasari commented on src/main/java/com/simibubi/create/content/redstone/displayLink/target/SignDisplayTarget.java:
```diff
@@ -16,26 +15,31 @@ public class SignDisplayTarget extends DisplayTarget {
 
 		boolean changed = false;
-		SignText signText = new SignText();
+		SignText signTextFront = ((SignBlockEntity) be).getFrontText();
+		SignText signTextBack = ((SignBlockEntity) be).getBackText();
```
Wouldn't that be better?

```suggestion
		SignText signTextFront = sign.getFrontText();
		SignText signTextBack = sign.getBackText();
```

Rafacasari replied:
Also you don't need {} in every if statement, just in the `if (changed) { ... }`

NickTheEASDude commented:
Does this also fix the issue where the sign's data was completely cleared when the text got updated?

dan96kid commented:
> Does this also fix the issue where the sign's data was completely cleared when the text got updated?

As far as I can tell, yes. This fix preserves both the text on the sign and any dyes that were applied.

dan96kid commented:
Looks like a solution to the problem this pull request is meant to fix has been committed: https://github.com/Creators-of-Create/Create/commit/b20c88ede2137c3cecf8278b30031162bf1ffa1d

Consider this pull request obsolete and closed.
